.github/workflows/pr-realm-js.yml

```
@@ -0,0 +1,306 @@
+
+      - name: Download prebuild artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: prebuild-darwin-x64
+          path: prebuilds
+
+      - name: Download xcframework artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: xcframework-macos
+          path: react-native/ios/realm-js-ios.xcframework
+
+      - name: Cache iOS build artifacts
+        id: cache-ios-build
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-ios-build
+        with:
+          path: react-native/ios/build
+          key: build-ios-${{ hashFiles('./src/**') }}
+          restore-keys: |
+            build-ios-
+
+      - name: Setup environment
+        run: |
+          brew install watchman
+
+      - name: Install project dependencies
+        run: |
+          npm ci
+
+      - name: Install test dependencies
+        working-directory: ./tests
+        run: |
+          npm ci
+
+      - name: Run react native tests
+        run: |
+          bash ./scripts/ci-scripts/react-tests.sh
+
+      - name: Run react native example
+        run: |
+          bash ./scripts/ci-scripts/realm-example.sh
+
+  test-react-native-android:
+    name: Test React Native on Android Emulator
+    runs-on: macos-latest
+    if: ${{false}}
+    needs: [build-and-test]
+    env:
+      REALM_DISABLE_SYNC_TESTS: true
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          submodules: "recursive"
+
+      - name: Download prebuild artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: prebuild-darwin-x64
+          path: prebuilds
+
+      - name: Download android artifact
+        uses: actions/download-artifact@v2
+        with:
+          name: android-artifact-x86_64
+          path: react-native/android/src/main/jniLibs
+
+      - name: Setup environment
+        run: |
+          brew install watchman
+
+      - name: Install project dependencies
+        run: |
+          npm ci
+
+      - name: Install test dependencies
+        working-directory: ./tests
+        run: |
+          npm ci
+
+      - name: run tests
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          api-level: 28
+          target: default
+          arch: x86_64
+          profile: Nexus 6
+          script: bash ./scripts/ci-scripts/test-android.sh
+
+  publish:
+    runs-on: ubuntu-latest
+    if: ${{false}}
+    needs: [build-and-test, test-react-native-ios, test-react-native-android]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Cache js build artifacts
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-js-build-artifacts
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{hashFiles('**/package-lock.json')}}
```

Feels redundant to have the `-build-` when we also have the `cache-name`.
```suggestion
          key: ${{ runner.os }}-${{ env.cache-name }}-${{hashFiles('**/package-lock.json')}}
```